.github/workflows/build_documentation.yml

```diff
@@ -0,0 +1,61 @@
+name: Build documentation
+
+on:
+  push:
+    branches:
+      - new_front_github_actions
+
+env:
+  HUGGINGFACE_PUSH: ${{ secrets.HUGGINGFACE_PUSH }}
+
+jobs:
+  build_and_package:
+    runs-on: ubuntu-latest
+    defaults:
+      run:
+        shell: bash -l {0}
+
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          repository: 'huggingface/doc-builder'
+          token: ${{ secrets.HUGGINGFACE_PUSH }}
+          ref: launch-script
+
+      - name: Clone transformers
+        run: |
+          git clone https://github.com/huggingface/transformers
+
+      - name: Setup environment
+        run: |
+          sudo apt-get -y update && sudo apt-get install -y libsndfile1-dev
+
+          pip install -e .
+          pip install git+https://github.com/huggingface/transformers#egg=transformers[dev]
+
+          export TORCH_VERSION=$(python -c "from torch import version; print(version.__version__.split('+')[0])")
+          pip install torch-scatter -f https://data.pyg.org/whl/torch-${TORCH_VERSION}+cpu.html
+
+          pip install torchvision
+          python -m pip install 'git+https://github.com/facebookresearch/detectron2.git'
+
+          sudo apt install tesseract-ocr
+          pip install pytesseract
+          pip install pytorch-quantization --extra-index-url https://pypi.ngc.nvidia.com
+
+      - name: Setup git
+        run: |
+          git config --global user.name HuggingFace
+          git config --global user.email transformers@huggingface.co
+          git checkout -b new-branch
+
+      - name: Make documentation
+        run: |
+          python src/build.py transformers ./transformers/docs/source
+
+      - name: Push to repository
+        run: |
+          git add build
+          git commit -m "Updated with commit ${{ github.sha }}"
+          git push --force origin new-branch
```

Comment:
The `--force` (at least on the `main` branch) would not be desirable on the hub side as it will make maintaining a sync'd local clone of `doc-builder` way harder

Is the issue that you expect this workflow to run with some concurrency?

Reply:
The `--force` would of course be removed before merge, it's just here for testing (I'm pushing to `new-branch`, but I'll push to `master` before merging)

Reply:
Ok!